packages/babel-parser/src/parser/statement.js

```diff
@@ -533,11 +547,13 @@ export default class StatementParser extends ExpressionParser {
 
     const refExpressionErrors = new ExpressionErrors();
     const init = this.parseExpression(true, refExpressionErrors);
-    if (this.match(tt._in) || this.isContextual("of")) {
+    const isForOf = this.isContextual("of");
+    if (this.match(tt._in) || isForOf) {
```

Comment:
This is a really minor optimization, but we can avoid the unnecessary `this.match` for `for-of` loops
```suggestion
    if (isForOf || this.match(tt._in)) {
```